test/functional/mongo_client_tests.js

```diff
@@ -219,7 +219,13 @@ describe('MongoClient', function() {
         return this.skip();
       }
       const replicaSetName = this.configuration.options.setName;
-      const client = configuration.newClient(this.configuration.options.url, {
+      var url = f(
+        'mongodb://127.0.0.1:%s/integration_test_?rs_name=%s',
```

Comment:
The test `MongoClient Should correctly pass through extra replicaset options fails` was previously failing only on Travis (not local instance of replicaset) with the error `No hostname or hostnames provided in connection string` but is now passing with this edit.

Reply:
What part of the test runner was causing this failure? Like I said before, I think we should really try to minimize changes to tests in this PR.

Reply:
@daprahamian thought it might have something to do with Travis not recognizing `localhost` as a hostname? I am not sure why this didn't work on Travis, since it worked on the local instance.

Reply:
This edit just changed the host in the URL from `localhost` into `127.0.0.1`, the `port` and the `rs_name` remain the same

Reply:
Reverted back to original way of using `configuration.url()` rather than `configuration.options.url`